IM_calculation/IM/Burks_Baker_2013_elastic_inelastic.py

```diff
@@ -67,8 +67,13 @@ def Bilinear_Newmark_withTH(
 
     # % Interpolate p=-ag*m (linearly)
     p = -ag * m
-    tg = np.linspace(0, ag.size - 1, ag.size) * dtg
-    t = np.linspace(0, tg[-1], int(tg[-1] / dt) + 1)  # num of steps increased from tg
+
+    tg = np.arange(
+        0, ag.size * dtg, dtg, dtype=float
+    )  # Creating [ 0, dtg, 2*dtg.... (ag.size-1)*dtg ].
+    t = np.arange(
+        0, ag.size * dtg, dt, dtype=float
+    )  # For the same begin and end, create a range spaced by dt. num of steps increased from tg
```

Comment:
I think this is a lot more intuitive than the original code. This is also more reliable we don't do the division to work out the number of steps (unlike linspace())

Reply:
Reading the documentation, using non integer step size with arange is often not consistent.
It may be better to calculate the number of timesteps and then multiply by the timestep size.

https://numpy.org/doc/stable/reference/generated/numpy.arange.html

Reply:
We may need to keep an eye on this going forward, but it should be fine

Reply:
To see why the original code is not correct, try:
ag.size = 7500, dtg=0.02, dt=0.005
tg.size is 7500, and t.size is 29996, while it should be 30000.
The second linspace could have been
`tg=np.linspace(0,(ag.size-1)*dtg,ag.size)
t=np.linspace(0,ag.size*dtg-dt,int(np.ceil((ag.size*dtg)/dt)))`
but as you can see, linspace() is not very readable - compare with the equivalent arange()

Reply:
It's because linkspace()'s second argument is the last element (inclusive). As we wanted ag.size*dtg to be the upperbound, we put (ag.size-1)*dtg for tg. For t, we also want ag.size*dtg to be the upperbound, but as the second argument is the last element, it should be ag.size*dtg-dt. If we want to use tg[-1] as in the original code, it should have been 
`
t=np.linspace(0,tg[-1]+dtg-dt, int(np.ceil( (tg[-1]+dtg)/dt)))
`

Reply:
@jpa198  You win. Came across a case where arange() fails and goes over the specified value of "stop". Reverted to linspace(), but left the arange() equivalents as comment to make it easier to understand